test/Gen.hs

```diff
@@ -105,9 +117,15 @@ s = genT
 type S = T "S"
 
 w :: Gen W
-w = list (linear 0 100) genT
+w = genT
+
+type W = T "W"
+
+fn :: (Fn.Arg a, Fn.Vary a, Show a) => Gen b -> Gen (a -> b)
+fn b = Gen (lift (fmap (fmap runTerm) . showingFn <$> Fn.fn (fst <$> runWriterT (runGen b))))
 
-type W = [T "W"]
+choice :: [Gen a] -> Gen a
+choice = Gen . Hedgehog.Gen.choice . Prelude.map runGen
```

Comment:
We don’t use many of hedgehog’s combinators in the property tests, but we do use `choice`, so I’ve wrapped it up.